File: frontend/src/app/shared/responses/portfolio-images.response.ts
export class PortfolioImagesResponse {

  images: PortfolioImage[];

  constructor(images: PortfolioImage[]) {
    this.images = images;
  }

}

export class PortfolioImage {

  id: number;
  name: string;

  constructor(id: number, name: string) {
    this.id = id;
    this.name = name;
  }

}

export class PortfolioImageWithBytes {
  id: number;
  name: string;
  bytes: string;

  constructor(id: number, name: string, bytes: string) {
    this.id = id;
    this.name = name;
    this.bytes = bytes;
  }
}

File: src/main/java/com/sgorokh/TimeSaver/controllers/responses/PortfolioImagesResponse.java
package com.sgorokh.TimeSaver.controllers.responses;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class PortfolioImagesResponse {

    private List<PortfolioImage> images = new ArrayList<>();

    public void addImage(Long id, String name) {
        images.add(new PortfolioImage(id, name));
    }

    @Data
    @AllArgsConstructor
    public static class PortfolioImage {

        private Long id;
        private String name;

    }
}
